core/modules/filters/kin.js

```diff
@@ -0,0 +1,107 @@
+/*\
+title: $:/core/modules/filters/kin.js
+type: application/javascript
+module-type: filteroperator
+
+Filter operator that recursively finds kindred between tiddlers
+
+\*/
+(function () {
+
+  /*jslint node: true, browser: true */
+  /*global $tw: false */
+  "use strict";
+
+  // TODO: Should I set global tw to true?
+
+  function collectFamilyMembers(tiddler, title, fieldname, direction) {
+    var family_members_from = [],
+      family_members_to = [];
+
+    function addToResultsIfNotFoundAlready(list, title) {
+      if (list.includes(title)) {
+        return false;
+      }
+      list.push(title);
+      return true
+    }
+
+    function findKindredFrom(tiddler, title) {
+      if (addToResultsIfNotFoundAlready(family_members_from, title)) {
+        if (tiddler) {
+          tiddler.getFieldList(fieldname).forEach(function (target_title) {
```

Comment:
Coding style: We avoid ES2015 `forEach` in favour of the `$tw.utils.each` method -- this is partly for performance, but mainly to retain compatibility with older JS platforms.